### PR TITLE
Datastructure

### PR DESCRIPTION
**Pull request by tvheems**
The data structure and the edited parser.

### CODE REVIEW

File: src/main/java/Parsing/Parser.java [hunk 1]
```diff
@@ -24,66 +34,72 @@ public static Parser getInstance() {
     }
```
Comment (MStraatman) [inline src/main/java/Parsing/Parser.java]: Nice use of singleton design pattern!

File: src/main/java/Parsing/Parser.java [hunk 2]
```diff
@@ -24,66 +34,72 @@ public static Parser getInstance() {
...
+
         try {
-            BufferedReader in = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream(filename)));
+            BufferedReader in = new BufferedReader(
+                    new InputStreamReader(getClass().getResourceAsStream(filename)));
             String line = in.readLine();
 
             while (line != null) {
-                System.out.println(line);
+//                System.out.println(line);
```
Comment (jjtiessen) [inline src/main/java/Parsing/Parser.java]: Remove commented code

File: src/main/java/datastructure/NodeDB.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            ResultSet rs = stmt.executeQuery();
+            rs.first();
+            int length = rs.getInt("length");
+
+            stmt = con.prepareStatement(
+                    "SELECT DISTINCT from_id FROM EDGES WHERE to_id = ?");
+            stmt.setInt(1, id);
+            rs = stmt.executeQuery();
+
+
```
Comment (jjtiessen) [inline src/main/java/datastructure/NodeDB.java]: Redundant empty line

File: src/main/java/datastructure/NodeDB.java [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        while (!nodeQueue.isEmpty()) {
+            int tempId = nodeQueue.poll();
+            if (!res.containsKey(tempId)) {
+                res.put(tempId, getNode(tempId));
+            }
+        }
+
+        return res;
+    }
+}
```
Comment (MStraatman) [inline src/main/java/datastructure/NodeDB.java]: Great preparation for subgraph queries.

File: src/main/java/datastructure/NodeDBTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    void setUp() {
+        nodedb = new NodeDB();
+    }
+
+    @org.junit.jupiter.api.AfterEach
+    void tearDown() {
+        nodedb = null;
+    }
+
+}
```
Comment (MStraatman) [inline src/main/java/datastructure/NodeDBTest.java]: There should be a newline at the end of the file.

File: src/main/java/datastructure/NodeTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @org.junit.jupiter.api.Test
+    void equals() {
+        assertTrue(node.equals(new Node(0, 0, new int[0], new int[0])));
+    }
+
+    @org.junit.jupiter.api.Test
+    void equalsNull() {
+        assertFalse(node.equals(null));
+    }
+}
```
Comment (MStraatman) [inline src/main/java/datastructure/NodeTest.java]: Newline at end...

File: src/main/java/datastructure/NodeDB.java [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+                icedges[i] = rs.getInt("from_id");
+                rs.next();
+            }
+
+            stmt = con.prepareStatement(
+                    "SELECT DISTINCT to_id FROM EDGES WHERE from_id = ?");
+            stmt.setInt(1, id);
+            rs = stmt.executeQuery();
+
+            rs.first();
```
Comment (jjtiessen) [inline src/main/java/datastructure/NodeDB.java]: I believe this should be removed.

File: src/main/java/datastructure/NodeDBTest.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    @org.junit.jupiter.api.BeforeEach
+    void setUp() {
+        nodedb = new NodeDB();
+    }
+
+    @org.junit.jupiter.api.AfterEach
+    void tearDown() {
+        nodedb = null;
+    }
+
```
Comment (jjtiessen) [inline src/main/java/datastructure/NodeDBTest.java]: This class does not actually test anything, right?

Comment (tvheems) [inline src/main/java/datastructure/NodeDBTest.java]: This is true, I deleted it.

File: src/main/java/datastructure/NodeDB.java (hunk 7 above)
Comment (tvheems) [inline src/main/java/datastructure/NodeDB.java]: It should, and is now.

File: src/main/java/Parsing/Parser.java (hunk 2 above)
Comment (tvheems) [inline src/main/java/Parsing/Parser.java]: I've deleted it.

File: src/main/java/datastructure/NodeDB.java (hunk 3 above)
Comment (tvheems) [inline src/main/java/datastructure/NodeDB.java]: Deleted the empty line

**Pull request merged**